db/mainphotos.sql

```diff
@@ -0,0 +1,80 @@
+DROP DATABASE IF EXISTS mainphotos;
+CREATE DATABASE mainphotos;
+
+\c mainphotos
+
+CREATE SCHEMA IF NOT EXISTS mainphotos;
+SET search_path TO mainphotos, public;
+
+CREATE TABLE mainphotos.neighborhoods (
+  neighborhood_id INT GENERATED BY DEFAULT AS IDENTITY NOT NULL PRIMARY KEY,
```

Comment:
you can use this shorthand:
- neighborhood_id SERIAL PRIMARY KEY,

**SERIAL** keyword does:
- auto increment
- add INT constraint
- add NOT NULL constraint
- assigns the owner of the sequence to the id column; as a result, the sequence object is deleted when the id column or table is dropped

Reply:
Serial does serve the same purpose as identity with auto-incrementing, but I think serial is less robust than identity in that serial is difficult to remove as a property from the table compared to identity due to its hack-y implementation in postgres and copies of the table will continue the sequence instead of resetting the count. Let me know if I'm understanding this wrong, I'm still trying to understand the nuances in postgres. 
Your point brings up an issue with my code though, I realized that by using GENERATED BY DEFAULT AS IDENTITY instead of GENERATED ALWAYS AS IDENTITY, I made it so that inserting rows manually could cause for errors since the internal incrementation doesn't account for manually defined values that are supposed to be auto-incremented. I'll change this in the next PR
Source: https://www.2ndquadrant.com/en/blog/postgresql-10-identity-columns/

Reply:
Yeah, it seems like SERIAL is the old way of doing auto-incrementing and the ids don't reset if a record is deleted. However, my understanding of what errors out when depends. Say you have a table with an existing record:
- `1, 'blue`

note the nuances if you do an insertion: 
- `insert (2, 'red')` will error out for GENERATED ALWAYS, but works for GENERATED BY DEFAULT
- `insert ('red')` works for both

which is to say that GENERATED ALWAYS is stricter in enforcing a user to NOT SPECIFY the ID when adding a record and more importantly, prevents a user from overriding data.

Hence my vote is GENERATED ALWAYS as well :)